ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Worker.java

```diff
@@ -531,29 +531,26 @@ protected Boolean findNextCompactionAndExecute(boolean computeStats) throws Inte
       final StatsUpdater su = computeStats ? StatsUpdater.init(ci, msc.findColumnsWithStats(
           CompactionInfo.compactionInfoToStruct(ci)), conf,
           runJobAsSelf(ci.runAs) ? ci.runAs : t.getOwner()) : null;
-      final CompactorMR mr = new CompactorMR();
+
       try {
-        if (runJobAsSelf(ci.runAs)) {
-          mr.run(conf, jobName.toString(), t, p, sd, tblValidWriteIds, ci, su, msc, dir);
+        failCompactionIfSetForTest();
```

Comment:
Is this test code?
Could we find another way to test this?

Reply:
Yes, it's test code and it used to be in CompactorMr#run, I just refactored it here.
Great question, shall I raise a jira for it?

Reply:
Yes, please raise a jira!

Reply:
Done: HIVE-24429

Reply:
Thanks @klcopp!